### PR TITLE
Derived sensors

### PR DESCRIPTION
**Pull request by Flix6x**
A currently undocumented feature to set up (using a new plugin config setting) additional FlexMeasures sensors whose data is derived from other sensors (that are directly mapped to a Z-info sensor).

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -129,25 +129,115 @@ def import_sensor_data(dryrun: bool = False):
...
+            ("shift", dict(periods=-1)),
+            ("head", dict(n=-1)),
+        ],
+    """
+    for method, kwargs in pandas_method_kwargs:
+        df = getattr(df, method)(**kwargs)
+    return df
+
+
+def ensure_zinfo_derived_sensors() -> List[Sensor]:
```
Comment (nhoening) [inline __init__.py]: I don't see (wihtout too much effort) what the difference in fucntionality is between this function and `ensure_zinfo_main_sensors`. Please say so in the docstring, and/or maybe refactor a bit if useful.

File: __init__.py [hunk 2]
```diff
@@ -129,25 +129,115 @@ def import_sensor_data(dryrun: bool = False):
...
+        sensor = (
+            Sensor.query.join(GenericAsset)
+            .filter(
+                Sensor.name == sensor_name,
+                Sensor.generic_asset_id == GenericAsset.id,
+                GenericAsset.name == generic_asset_name,
+            )
+            .one_or_none()
+        )
+        if sensor is None:
```
Comment (nhoening) [inline __init__.py:225]: For instance, this whole block seems identical to the block in the other function?

**Review by nhoening — CHANGES_REQUESTED** [review]

File: __init__.py (hunk 1 above)
Comment (Flix6x) [reply __init__.py]: I now refactored the config setting to streamline the two different ways I had to loop through the main and derived sensor descriptions.

**Comment by nhoening** [comment]
So the derived sensors - are they really z-info sensors? It seems to me they are derived from Z-Info sensors, but really not a representation of any z-info raw data. 
So the naming should maybe stick to DERIVED_SENSORS or similar...

**Comment by Flix6x** [comment]
I made a [ticket](https://github.com/SeitaBV/flexmeasures-zinfo/issues/4) to refactor further.

**Comment by Flix6x** [comment]
> So the derived sensors - are they really z-info sensors? It seems to me they are derived from Z-Info sensors, but really not a representation of any z-info raw data.
> So the naming should maybe stick to DERIVED_SENSORS or similar...

It's probably not even necessary to make the distinction once we move to a single config setting. Perhaps the [unified config setting](https://github.com/SeitaBV/flexmeasures-zinfo/issues/4) could be renamed again to `ZINFO_SENSOR_MAPPING`.

**Review by nhoening — APPROVED** [review]

**Pull request merged**